core/src/services/icloud/core.rs

```diff
@@ -489,9 +527,12 @@ impl PathQuery for IcloudPathQuery {
 
     async fn create_dir(&self, parent_id: &str, name: &str) -> Result<String> {
         let mut signer = self.signer.lock().await;
-        let client_id = signer.client_id();
+        let clone = signer.clone();
```

Comment:
Why we need this? Maybe we can change `client_id()`'s API to `&mut self` instead.